dojo/problem/redis.py

```diff
@@ -0,0 +1,155 @@
+            finding_ids=set(data.get("finding_ids", [])),
+        )
+
+
+@lru_cache(maxsize=1)
+def get_redis_client():
+    return redis.Redis(host="django-defectdojo-redis-1", port=6379, decode_responses=True)
+
+
+def add_finding_to_redis(finding):
+    if finding.vuln_id_from_tool and finding.severity != "Info":
+        redis_client = get_redis_client()
+        if redis_client.exists("problems", "id_to_problem"):
+            script_id = finding.vuln_id_from_tool
+            script_to_problem_mapping = problems_help.load_json()
+            problem_id = script_to_problem_mapping.get(script_id, script_id)
+            problem_id = base64.b64encode(problem_id.encode()).decode()
+            problem = redis_client.hget("problems", problem_id)
+            if problem:
+                problem = Problem.from_dict(json.loads(problem))
+                problem.finding_ids.add(finding.id)
+                problem.script_ids.add(finding.vuln_id_from_tool)
+                if SEVERITY_ORDER[finding.severity] > SEVERITY_ORDER[problem.severity]:
+                    problem.name = finding.title
+                    problem.severity = finding.severity
+                redis_client.hset("problems", problem_id, json.dumps(problem.to_dict()))
+                redis_client.hset("id_to_problem", finding.id, problem_id)
+            else:
+                problem = Problem(finding.title, problem_id, finding.severity)
+                problem.finding_ids.add(finding.id)
+                problem.script_ids.add(finding.vuln_id_from_tool)
+                redis_client.hset("problems", problem_id, json.dumps(problem.to_dict()))
+                redis_client.hset("id_to_problem", finding.id, problem_id)
+        else:
+            dict_problems_findings()
+
+
+def remove_finding_from_redis(finding_id_to_remove):
+    redis_client = get_redis_client()
+    if redis_client.exists("problems", "id_to_problem"):
+        problem_id = redis_client.hget("id_to_problem", finding_id_to_remove)
+        if problem_id:
+            problem = Problem.from_dict(json.loads(redis_client.hget("problems", problem_id)))
+            problem.finding_ids.remove(finding_id_to_remove)
+            if not problem.finding_ids:
+                redis_client.hdel("problems", problem_id)
+            else:
+                findings = Finding.objects.filter(id__in=problem.finding_ids)
+                severity, name, script_ids = "Info", "", set()
+                for finding in findings:
+                    script_ids.add(finding.vuln_id_from_tool)
+                    if SEVERITY_ORDER[finding.severity] > SEVERITY_ORDER[severity]:
+                        name = finding.title
+                        severity = finding.severity
+                problem.name = name
+                problem.severity = severity
+                problem.script_ids = script_ids
+                redis_client.hset("problems", problem_id, json.dumps(problem.to_dict()))
```

Comment:
Can we instead encapsulate this and just call functions here? Something like

```python
problem = Problem.load_from_id(problem_id, redis_client)
problem.remove_finding(finding_id_to_remove)
problem.persist(redis_client)
```

The problem with the current approach is that I have to remember how data is stored on Redis every time... because it's spread/repeated in many places.